core/src/main/java/bisq/core/dao/node/lite/LiteNode.java

```diff
@@ -78,6 +86,31 @@ public void start() {
         super.onInitialized();
 
         liteNodeNetworkService.start();
+
+        bsqWalletService.addNewBestBlockListener(block -> {
+            int height = block.getHeight();
+            log.info("New block at height {} from bsqWalletService", height);
+
+            // Check if we are done with parsing
+            if (!daoStateService.isParseBlockChainComplete())
+                return;
+
+            if (checkForBlockReceivedTimer != null) {
+                // In case we received a new block before out timer gets called we stop the old timer
+                checkForBlockReceivedTimer.stop();
+            }
+
+            // We expect to receive the new BSQ block from the network shortly after BitcoinJ has been aware of it.
+            // If we don't receive it we request it manually from seed nodes
+            checkForBlockReceivedTimer = UserThread.runAfter(() -> {
+                if (daoStateService.getChainHeight() < height) {
+                    log.warn("We did not receive a block from the network {} seconds after we saw the new block in BicoinJ. " +
+                                    "We request from our seed nodes missing blocks from block height {}.",
+                            CHECK_FOR_BLOCK_RECEIVED_DELAY_SEC, daoStateService.getChainHeight());
+                    liteNodeNetworkService.requestBlocks(daoStateService.getChainHeight());
```

Comment:
This means we're asking for block `daoStateService.getChainHeight()` and up, but we already have the block at `daoStateService.getChainHeight()`, wouldn't it be better to ask for `daoStateService.getChainHeight() + 1`?

It's also only asking seednodes, any full node should be fine to ask and then we can use the block hash to verify we have the right blocks, and if not we block that ban that full node and ask the next. That's not really relevant to this PR though.

Reply:
I think you are right, thanks! Will look into it again after I finish current new PR...

Reply:
Fixed